auditbeat/module/file_integrity/file_parsers.go

```diff
@@ -0,0 +1,123 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package file_integrity
+
+import (
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+// FileParser is a file analyser the provides inrichment for file.* fields.
+type FileParser interface {
+	Parse(dst common.MapStr, path string) error
+}
+
+// FileParsers returns the set of file parsers required to satisfy the config.
+func FileParsers(c Config) []FileParser {
+	// TODO: Consider whether to allow specification by fileparser name in
+	// addition to target field.
+	// TODO: Implement globbing paths so that the user can write things like
+	// "file.*.imports".
```

Comment:
Probably the best way to do this is to expand a RE2 pattern over the parser-available fields rather than inventing a new incomplete globbing syntax. The world already has enough of those.